### PR TITLE
[#4433] Improvement(catalog-lakehouse-paimon): Update Paimon Catalog doc

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?
Update Paimon Catalog doc, add the `AlterTable`, `partition` and `Index` related docs.

### Why are the changes needed?

Fix: https://github.com/apache/gravitino/issues/4433

### CODE REVIEW

**Comment by caican00** [comment]
cc @FANNG1

File: docs/lakehouse-paimon-catalog.md [hunk 1]
```diff
@@ -72,17 +71,46 @@ Please refer to [Manage Relational Metadata Using Gravitino](./manage-relational
...
-- Supporting createTable, dropTable, loadTable and listTable.
+- Supporting createTable, dropTable, alterTable, loadTable and listTable.
 ```
 dropTable will delete the table location directly, similar with purgeTable.
 ```
-- Supporting Column default value through table properties, such as `fields.{columnName}.default-value`.
+- Supporting Column default value through table properties, such as `fields.{columnName}.default-value`, not column expression.
 
-- Doesn't support alterTable now.
+:::caution
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:80]: info?

File: docs/lakehouse-paimon-catalog.md [hunk 2]
```diff
@@ -94,7 +122,13 @@ dropTable will delete the table location directly, similar with purgeTable.
 
 ### Table indexes
 
-- Doesn't support table indexes.
+- Only supports primary key Index.
+
+:::info We cannot specify more than one primary key Index, and a primary key Index can contain multiple fields as a joint primary key.
+:::
+
+:::caution Paimon Table primary key constraint should not be same with partition fields, this will result in only one record in a partition.
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:130]: info?

File: docs/lakehouse-paimon-catalog.md (hunk 1 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: updated.

File: docs/lakehouse-paimon-catalog.md [hunk 3]
```diff
@@ -94,7 +122,13 @@ dropTable will delete the table location directly, similar with purgeTable.
 
 ### Table indexes
 
-- Doesn't support table indexes.
+- Only supports primary key Index.
+
+:::info We cannot specify more than one primary key Index, and a primary key Index can contain multiple fields as a joint primary key.
```
[the remark this answers is not in the document]
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: updated

File: docs/lakehouse-paimon-catalog.md (hunk 2 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: updated.

File: docs/lakehouse-paimon-catalog.md [hunk 4]
```diff
@@ -72,17 +71,46 @@ Please refer to [Manage Relational Metadata Using Gravitino](./manage-relational
...
-- Doesn't support alterTable now.
+:::caution
+Paimon does not support auto increment column.
+:::
+
+#### Table changes
+
+- AddColumn
+```
+AddColumn supports add a new column with first, after and defualt position.
```
[the remark this answers is not in the document]
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: removed.

**Comment by caican00** [comment]
comments have been addressed. @FANNG1

File: docs/lakehouse-paimon-catalog.md [hunk 5]
```diff
@@ -94,7 +119,17 @@ dropTable will delete the table location directly, similar with purgeTable.
 
 ### Table indexes
 
-- Doesn't support table indexes.
+- Only supports primary key Index.
+
+:::info
+We cannot specify more than one primary key Index,
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:125]: place in one line?

File: docs/lakehouse-paimon-catalog.md [hunk 6]
```diff
@@ -94,7 +119,17 @@ dropTable will delete the table location directly, similar with purgeTable.
...
+and a primary key Index can contain multiple fields as a joint primary key.
+:::
+
+:::info
+Paimon Table primary key constraint should not be same with partition fields,
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:130]: place in one line?

**Review by FANNG1 — APPROVED** [review]

File: docs/lakehouse-paimon-catalog.md (hunk 6 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: fixed

File: docs/lakehouse-paimon-catalog.md (hunk 5 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: fixed

**Pull request merged**